docs: prepare Balances for GA (#679)
tkhq/docs: snippets/shared/balance-concepts.mdx

### Webhooks

Turnkey Webhooks let you react to balance changes in real time, without polling. Instead of repeatedly calling the Get Balances API to check for updates, you register an endpoint and Turnkey pushes the data to you. Each payload carries the balance diff for that event (asset, address, and amount transferred), so your application can update state immediately without an additional API call. Combined with the Balance APIs, you have everything you need to keep your application's balances up to date without building your own indexing/polling infrastructure or relying on another third party.

You subscribe to webhooks at the parent organization level. Subscriptions cover wallet-account addresses across the parent organization and all of its sub-organizations. In other words, once you subscribe you'll receive webhook notifications for all the addresses within your entire Turnkey instance.

> #### The two-webhook model

Reviewer: I might change this header to something like this:

```suggestion
#### Webhooks - Confirmed vs. Finalized
```

Turnkey emits two event types for every balance change:

- `"balances:confirmed"` fires when a supported asset transfer is first included in a block onchain. The transfer has happened, but a chain reorganization could still remove it from the canonical chain.
- `"balances:finalized"` fires once the containing block has reached the finalization threshold for that chain. At this point the chance of the transfer being removed by a reorg is negligible.

> You can subscribe to one or both. For example, a consumer app that wants its UI to react immediately can listen on `"balances:confirmed"`, render the deposit as pending, and clear that state when the matching `"balances:finalized"` arrives. An application moving high-value transfers that needs certainty before crediting funds can ignore `"balances:confirmed"` entirely and act only on `"balances:finalized"`.

Reviewer: Love this explanation!

#### Subscribing

> Use the [Create Webhook Endpoint](/api-reference/activities/create-webhook-endpoint) API with the `BALANCE_CONFIRMED_UPDATES` or `BALANCE_FINALIZED_UPDATES` event type. You can subscribe to one or both by registering a subscription per event type. The endpoint itself is created on the parent organization. Once active, the subscription delivers events for wallet-account addresses across the parent organization and all of its sub-organizations. This covers both addresses Turnkey generated and addresses imported into your Turnkey organization.

Reviewer: Some minor fixes:
```suggestion
Use the [Create Webhook Endpoint](/api-reference/activities/create-webhook-endpoint) API with the `BALANCE_CONFIRMED_UPDATES` or `BALANCE_FINALIZED_UPDATES` event type. You can subscribe to one or both by registering a subscription per event type. The endpoint itself is associated with the parent organization. Once active, the events will be delivered for wallet-account addresses across the parent organization and all of its sub-organizations. This covers both Turnkey-generated addresses and imported addresses.
```

#### Reorg handling

> The two-webhook model handles chain reorganizations without any rollback logic on your side. If a confirmed transaction is reorged out of the canonical chain before finality:

Reviewer: I might suggest the following changes (but you can take it or leave it :)):
```suggestion
The two-webhook model handles chain reorganizations transparently. If a confirmed transaction is involved in a reorg, the following sequence of webhooks will be seen:
```

1. The original `"balances:confirmed"` fires when the transaction is first included.
2. The reorg occurs and the transaction is removed from the canonical chain.
3. A new `"balances:confirmed"` fires when the transaction is included again in the new canonical chain.
4. A `"balances:finalized"` fires for the canonical inclusion once it reaches finality.

> If the transaction never re-enters the canonical chain, no `"balances:finalized"` event fires for it. Either way, `"balances:finalized"` is the source of truth for what actually landed onchain. You don't need to track and roll back state from earlier confirmed events.

Reviewer: Fantastic explanation 👍 .

#### Delivery

> Balance webhooks use at-least-once processing. If your endpoint is unreachable or a delivery fails on our side, Turnkey will retry. Each payload includes a per-event `idempotencyKey`, so you can safely deduplicate any duplicate deliveries on your end.

Reviewer: I'd suggest adding a small clarification for the `idempotencyKey`:
```suggestion
Balance webhooks use at-least-once processing. If your endpoint is unreachable or a delivery fails on our side, Turnkey will retry. Each webhook payload includes a `idempotencyKey` which is guaranteed to be unique, and which can be safely used to deduplicate deliveries on your end.
```
